From 41117d71bb58e001f6a2b6a11c9314d5b70b9182 Mon Sep 17 00:00:00 2001
From: Ben Gamari <ben@smart-cactus.org>
Date: Thu, 23 Jan 2020 14:11:20 -0500
Subject: [PATCH] base: Use one-shot kqueue on macOS

The underlying reason requiring that one-shot usage be disabled (#13903)
has been fixed.

Closes #15768.
---
 libraries/base/GHC/Event/Manager.hs | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/libraries/base/GHC/Event/Manager.hs b/libraries/base/GHC/Event/Manager.hs
index eda3e61490..51306795fe 100644
--- a/libraries/base/GHC/Event/Manager.hs
+++ b/libraries/base/GHC/Event/Manager.hs
@@ -150,9 +150,7 @@ callbackTableVar mgr fd = emFds mgr ! hashFd fd
 
 haveOneShot :: Bool
 {-# INLINE haveOneShot #-}
-#if defined(darwin_HOST_OS) || defined(ios_HOST_OS)
-haveOneShot = False
-#elif defined(HAVE_EPOLL) || defined(HAVE_KQUEUE)
+#if defined(HAVE_EPOLL) || defined(HAVE_KQUEUE)
 haveOneShot = True
 #else
 haveOneShot = False
@@ -365,20 +363,9 @@ registerFd mgr cb fd evs lt = do
   return r
 {-# INLINE registerFd #-}
 
-{-
-    Building GHC with parallel IO manager on Mac freezes when
-    compiling the dph libraries in the phase 2. As workaround, we
-    don't use oneshot and we wake up an IO manager on Mac every time
-    when we register an event.
-
-    For more information, please read:
-        https://gitlab.haskell.org/ghc/ghc/issues/7651
--}
 -- | Wake up the event manager.
 wakeManager :: EventManager -> IO ()
-#if defined(darwin_HOST_OS) || defined(ios_HOST_OS)
-wakeManager mgr = sendWakeup (emControl mgr)
-#elif defined(HAVE_EPOLL) || defined(HAVE_KQUEUE)
+#if defined(HAVE_EPOLL) || defined(HAVE_KQUEUE)
 wakeManager _ = return ()
 #else
 wakeManager mgr = sendWakeup (emControl mgr)
-- 
2.25.0
